### PR TITLE
add handleSwitchChain func

### PR DESCRIPTION
**Pull request by fritzschoff**
Clem did that for Aelin and was encouraging us to do the same. I thought that it would be a good addition to the provider package.
Code used in Aelin
```
const handleSwitchChain = useCallback(async () => {
        if (!provider || !network?.id || !walletAddress) return;
        const web3Provider = provider as ethers.providers.Web3Provider;
        if (!web3Provider.provider || !web3Provider.provider.request) return;
        const newNetworkId = getCorrespondingNetwork(network?.id, isOVM);
        const formattedChainId = ethers.utils.hexStripZeros(BigNumber.from(newNetworkId).toHexString());
        await web3Provider.provider.request({
            method: 'wallet_switchEthereumChain',
            params: [{ chainId: formattedChainId }],
        });
    }, [isOVM, provider, network?.id, walletAddress]);
```

⚠️ ⚠️ ⚠️
Made some test and it seems only the Coinbase wallet and MetaMask are supporting that request. Tried with the trust wallet and nothing happened in the app

### CODE REVIEW

File: packages/providers/src/types.ts [hunk 1]
```diff
@@ -13,3 +13,15 @@ type L1Provider = ethers.providers.Web3Provider | ethers.providers.InfuraProvide
 type L2Provider = OvmProvider;
 
 export type SynthetixProvider = L1Provider | L2Provider;
+
+export enum NetworkId {
```
Comment (fritzschoff) [inline packages/providers/src/types.ts:17]: would reuse the enum from the contracts-interface module, but if I import that module in the provider module, lerna is not symlinking the dependency and the build is crashing!

Comment (fritzschoff) [reply packages/providers/src/types.ts:17]: Error is: can't resolve @synthetixio/contracts-interface

**Comment by bachstatter** [comment]
Since they are different packages if you want to use something you need to install that package as a dependency.
Added the contract-interface as deps so we have access to `NetworkId`

**Review by bachstatter — APPROVED** [review]

**Comment by bachstatter** [comment]
Before merging you can install this version and make sure it behaves as you expect: `npm i @synthetixio/providers@0.0.0-5019e64`

**Pull request merged**